src/script/components/LexicalInput/utils/mention-utils.ts

```diff
@@ -0,0 +1,204 @@
+  `)`;
+
+export const LENGTH_LIMIT = 75;
+
+// Regex used to trigger the mention menu.
+function createMentionsRegex(triggers: string[], allowSpaces: boolean) {
+  return new RegExp(
+    `(^|\\s|\\()(${TRIGGERS(triggers)}((?:${
+      VALID_CHARS(triggers) + (allowSpaces ? VALID_JOINS : '')
+    }){0,${LENGTH_LIMIT}})` + `)$`,
+  );
+}
+
+export function checkForMentions(text: string, triggers: string[], allowSpaces: boolean): MenuTextMatch | null {
+  const match = createMentionsRegex(triggers, allowSpaces).exec(text);
+  if (match !== null) {
+    // The strategy ignores leading whitespace but we need to know it's
+    // length to add it to the leadOffset
+    const maybeLeadingWhitespace = match[1];
+    const matchingStringWithTrigger = match[2];
+    const matchingString = match[3];
+    if (matchingStringWithTrigger.length >= 1) {
+      return {
+        leadOffset: match.index + maybeLeadingWhitespace.length,
+        matchingString: matchingString,
+        replaceableString: matchingStringWithTrigger,
+      };
+    }
+  }
+  return null;
+}
+
+export function isWordChar(char: string, triggers: string[]) {
+  return new RegExp(VALID_CHARS(triggers)).test(char);
+}
+
+type SelectionInfo = {
+  node: LexicalNode;
+  offset: number;
+  isTextNode: boolean;
+  textContent: string;
+  selection: RangeSelection;
+  prevNode: LexicalNode | null;
+  nextNode: LexicalNode | null;
+  cursorAtStartOfNode: boolean;
+  cursorAtEndOfNode: boolean;
+  wordCharBeforeCursor: boolean;
+  wordCharAfterCursor: boolean;
+};
+
+export function getSelectionInfo(triggers: string[]): SelectionInfo | undefined {
+  const selection = $getSelection();
+  if (!selection || !$isRangeSelection(selection)) {
+    return undefined;
+  }
+
+  const anchor = selection.anchor;
+  const focus = selection.focus;
+  const nodes = selection.getNodes();
+  if (anchor.key !== focus.key || anchor.offset !== focus.offset || nodes.length === 0) {
+    return undefined;
+  }
+
+  const [node] = nodes;
+  const isTextNode = $isTextNode(node) && node.isSimpleText();
+  const offset = anchor.type === 'text' ? anchor.offset : 0;
+  const textContent = node.getTextContent();
+  const cursorAtStartOfNode = offset === 0;
+  const cursorAtEndOfNode = textContent.length === offset;
+  const charBeforeCursor = textContent.charAt(offset - 1);
+  const charAfterCursor = textContent.charAt(offset);
+  const wordCharBeforeCursor = isWordChar(charBeforeCursor, triggers);
+  const wordCharAfterCursor = isWordChar(charAfterCursor, triggers);
+  const prevNode = getPreviousSibling(node);
+  const nextNode = getNextSibling(node);
+
+  return {
+    node,
+    offset,
+    isTextNode,
+    textContent,
+    selection,
+    prevNode,
+    nextNode,
+    cursorAtStartOfNode,
+    cursorAtEndOfNode,
+    wordCharBeforeCursor,
+    wordCharAfterCursor,
+  };
+}
+
+export function insertMention(triggers: string[], trigger: string, value?: string) {
```

Comment:
To make the insertMention function more concise and readable, we can rewrite it like this.

```
// Insert a mention node or a text node with the trigger to open the mention menu.
  const mentionNode = value ? $createBeautifulMentionNode(trigger, value) : $createTextNode(trigger);

  // Insert a mention with a leading space if the node at the cursor is not a text node.
  if (!isTextNodeAtCursor(node, offset)) {
    insertWithSpace(node, mentionNode, $createTextNode(' '));
    return true;
  }

  const nodes: LexicalNode[] = [];
  if (shouldInsertSpaceBefore(node, offset, prevNode)) {
    nodes.push($createTextNode(' '));
  }
  nodes.push(mentionNode);
  if (shouldInsertSpaceAfter(node, offset, nextNode)) {
    const spaceNode = $createTextNode(' ');
    nodes.push(spaceNode);
    spaceNode.selectPrevious();
  } 

  return true;

// write those functions here
```